src/main/java/com/rybka/command/ConvertCommand.java

```diff
@@ -0,0 +1,79 @@
+package com.rybka.command;
+
+import com.rybka.dao.CurrencyHistoryDAO;
+import com.rybka.exception.CurrencyAPICallException;
+import com.rybka.exception.DBConnectionException;
+import com.rybka.model.CurrencyHistory;
+import com.rybka.model.UserConvertData;
+import com.rybka.service.exchange.ExchangeService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+@Log4j
+@RequiredArgsConstructor
+public class ConvertCommand implements Command {
+    private final Scanner scanner;
+    private final ExchangeService service;
+    private final CurrencyHistoryDAO currencyHistoryDAO;
+
+    @Override
+    public void execute() {
+        try {
+            System.out.print("Insert base currency: ");
```

Comment:
Better to move this logic to `readUserParameters` method.

Reply:
fixed